File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] charCount =  new int[128];
        
        int i=0;
        int j=0;
        int res=0;
        
        while(j<s.length()){
            char valR=s.charAt(j);
            charCount[valR]++;
            
            while(charCount[valR]>1){
             char valL=s.charAt(i);
             charCount[valL]--;
             i++;
            }
            res=Math.max(res,j-i+1);
            j++;
        }   
        return res;
    }
}

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start=0
        maxLen=-math.inf
        if len(s)==len(set(s)):
            return len(s)
        repeat={}
        for i in range(len(s)):
            if s[i] in repeat:
                start=max(start,repeat[s[i]]+1)
            maxLen=max(maxLen,i-start+1)
               
            repeat[s[i]]=i
        return maxLen